### ISSUE TITLE
Building on Windows x64

### ISSUE THREAD

**Issue by stakach** [joyent/libuv#1072]
Having trouble building libuv for Windows x64

Running command `vcbuild.bat release x64` leads to:

``` shell
X:\Documents\GitHub\libuv\ext\libuv\uv.sln.metaproj : error MSB4126: The specified solution configuration "Release|amd64" is invalid.
Please specify a valid solution configuration using the Configuration and Platform properties (e.g. MSBuild.exe Solution.sln /p:Configuration=Debug /p:Platform="Any CPU") or leave those properties blank to use the default solution configuration.
[X:\Documents\GitHub\libuv\ext\libuv\uv.sln]
```

Tried running on CMD and VS CMD
tested with VS 2010 and 2013

**Comment by saghul** [comment]
@stakach can you try https://github.com/joyent/libuv/pull/1108 ?

**Comment by stakach** [comment]
I'm getting this with a fresh install of VS2013 Express on a clean windows box

```
HEAD is now at db3ab6d build, windows: fixed x64 configuration issue

C:\Users\Steve\Desktop\libuv>vcbuild.bat release x64
git clone https://git.chromium.org/external/gyp.git build/gyp
Cloning into 'build/gyp'...
remote: Counting objects: 14601, done.
remote: Compressing objects: 100% (3042/3042), done.
remote: Total 14601 (delta 9977), reused 14589 (delta 9971)
Receiving objects: 100% (14601/14601), 7.99 MiB | 549.00 KiB/s, done.
Resolving deltas: 100% (9977/9977), done.
Checking connectivity... done.
Checking out files: 100% (1261/1261), done.
['-Dtarget_arch=x64', '-Duv_library=static_library', '.\\uv.gyp', '-I', '.\\comm
on.gypi', '--depth=.', '-Dhost_arch=AMD64', '-Dcomponent=static_library']
Project files generated.
Building the projects in this solution one at a time. To enable parallel build,
please add the "/m" switch.
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V120\Microsoft.Cpp.Platform.t
argets(64,5): error MSB8020: The build tools for Visual Studio 2010 (Platform T
oolset = 'v100') cannot be found. To build using the v100 build tools, please i
nstall Visual Studio 2010 build tools.  Alternatively, you may upgrade to the c
urrent Visual Studio tools by selecting the Project menu or right-click the sol
ution, and then selecting "Upgrade Solution...". [C:\Users\Steve\Desktop\libuv\
libuv.vcxproj]
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V120\Microsoft.Cpp.Platform.t
argets(64,5): error MSB8020: The build tools for Visual Studio 2010 (Platform T
oolset = 'v100') cannot be found. To build using the v100 build tools, please i
nstall Visual Studio 2010 build tools.  Alternatively, you may upgrade to the c
urrent Visual Studio tools by selecting the Project menu or right-click the sol
ution, and then selecting "Upgrade Solution...". [C:\Users\Steve\Desktop\libuv\
uv_dtrace_header.vcxproj]
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V120\Microsoft.Cpp.Platform.t
argets(64,5): error MSB8020: The build tools for Visual Studio 2010 (Platform T
oolset = 'v100') cannot be found. To build using the v100 build tools, please i
nstall Visual Studio 2010 build tools.  Alternatively, you may upgrade to the c
urrent Visual Studio tools by selecting the Project menu or right-click the sol
ution, and then selecting "Upgrade Solution...". [C:\Users\Steve\Desktop\libuv\
uv_dtrace_provider.vcxproj]
```

**Comment by saghul** [comment]
I was under the impression that you cannot create an x64 build with VS Express, can you?

**Comment by schlamar** [comment]
Looks like gyp didn't built the project for VS 2013. Maybe there is an issue with #1043. @rioki any idea? 

Interestingly, there is a 64bit compiler on VS Express 2010. Only the batch file setting the correct evironment variables is missing (`C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\bin\amd64\vcvars64.bat`). This means vcbuild.bat will fail but compiling directly in the IDE works.

And it looks like that in VS Express 2013 this is working now. Otherwise there should an error message like `The specified configuration type is missing. The tools for the configuration might not be installed.`.

**Comment by saghul** [comment]
> Looks like gyp didn't built the project for VS 2013. Maybe there is an issue with #1043. @rioki any idea? 

AFAIK it works just fine for x86 :-O

@stakach any chance you can test with with VS Express 2013? If the 2010 version is lacking that file, well, too bad, there is not much we can do, is there?

**Comment by stakach** [comment]
@schlamar I've definitely been using VS 2013 since #1043 without issue. I hadn't tested x64 as my windows development machine had 32bit binaries and all other tests were run on OSX / Travis.

Only when I had to do a Windows deploy and installed x64 binaries did I run into issue. I'll play around some more tomorrow on the staging environment I setup today.

**Comment by stakach** [comment]
@saghul that was VS 2013 output.

**Comment by retep998** [comment]
For the record, I am building libuv with my own custom project files using VS 2013 for x64 and it works just fine.

**Comment by stakach** [comment]
@retep998 no doubt! Just the auto-build that is broken - unless that works for you?

**Comment by retep998** [comment]
I haven't tried libuv's build system.

**Comment by schlamar** [comment]
@stakach What do you get when you open the generated uv.sln with VS2013. Does it open without error? Can you compile?

**Comment by stakach** [comment]
It appears it is building the project for VS2010
The project opens without issue however requires an upgrade before it will compile
![image](https://f.cloud.github.com/assets/368013/2167665/1d97520e-9520-11e3-9f4f-71349097632f.png)

If I request to upgrade the project to VS2013 then everything compiles. Build successful with a few more warnings than the x86 version.
![image](https://f.cloud.github.com/assets/368013/2167626/b86c5556-951e-11e3-86bf-9ede9d821ba5.png)

**Comment by schlamar** [comment]
I guess the VS 2013 detection does somehow not properly work on your new environment or does the x86 compilation work on this setup? Output of `SET` might be helpful to debug this.

**Comment by saghul** [comment]
Can you try to source the prober vcvarsall file and then start the build process? With a change I merged recently it shouldn't override it so it should use what's already defined.

**Comment by saghul** [comment]
Also, unfortunately, I don't have a 64bit Windows to test this myself, sorry.

**Comment by stakach** [comment]
`"%VS120COMNTOOLS%\..\..\vc\vcvarsall.bat"` is the correct location.

I've also found that if I run `vcbuild.bat release ia32` followed by `vcbuild.bat release x64` the x64 build succeeds.

x64 builds should work on 32bit windows too.

**Comment by saghul** [comment]
Hi, I tried to build it using VS 2013 express + the platform SDK and it won't let me. It fails saying that it cannot find msbuild. Should I be able to make a x64 build on a x86 system?

**Comment by stakach** [comment]
@saghul hmm I would assume you should be. VS 2013 Express is 32bit and cross compiles to x64 on my system and I don't have a 32bit system to test on..

Anyway at this point I think this pull request should be accepted. It is an issue with gyp: https://code.google.com/p/gyp/issues/detail?id=186&q=x64

**Comment by saghul** [comment]
@stakach ok, will do.

**Comment by saghul** [comment]
Landed #1108 in https://github.com/joyent/libuv/commit/8010bf9dd2a649bbe1e65b59e7bbd0e6bedd80ba
